instance/import.go

```diff
@@ -0,0 +1,187 @@
+				} else {
+					if err := s.UpdateImportObservationsTaskState(instanceID, tasks.ImportObservations.State); err != nil {
+						log.ErrorCtx(ctx, errors.WithMessage(err, "Failed to update import observations task state"), logData)
+						return &taskError{err, http.StatusInternalServerError}
+					}
+				}
+			} else {
+				validationErrs = append(validationErrs, errors.New("bad request - invalid import observation task, must include state"))
+			}
+		}
+
+		if tasks.BuildHierarchyTasks != nil {
+			hasImportTasks = true
+			var hasHierarchyImportTask bool
+			for _, task := range tasks.BuildHierarchyTasks {
+				hasHierarchyImportTask = true
+				if err := models.ValidateImportTask(task.GenericTaskDetails); err != nil {
+					validationErrs = append(validationErrs, err)
+				} else {
+					if err := s.UpdateBuildHierarchyTaskState(instanceID, task.DimensionName, task.State); err != nil {
+						if err.Error() == "not found" {
+							notFoundErr := task.DimensionName + " hierarchy import task does not exist"
+							log.ErrorCtx(ctx, errors.WithMessage(err, notFoundErr), logData)
+							return &taskError{errors.New(notFoundErr), http.StatusNotFound}
+						}
+						log.ErrorCtx(ctx, errors.WithMessage(err, "failed to update build hierarchy task state"), logData)
+						return &taskError{err, http.StatusInternalServerError}
+					}
+				}
+			}
+			if !hasHierarchyImportTask {
+				validationErrs = append(validationErrs, errors.New("bad request - missing hierarchy task"))
+			}
+		}
+
+		if tasks.BuildSearchIndexTasks != nil {
+			hasImportTasks = true
+			var hasSearchIndexImportTask bool
+			for _, task := range tasks.BuildSearchIndexTasks {
+				hasSearchIndexImportTask = true
+				if err := models.ValidateImportTask(task.GenericTaskDetails); err != nil {
+					validationErrs = append(validationErrs, err)
+				} else {
+					if err := s.UpdateBuildSearchTaskState(instanceID, task.DimensionName, task.State); err != nil {
+						if err.Error() == "not found" {
+							notFoundErr := task.DimensionName + " search index import task does not exist"
+							log.ErrorCtx(ctx, errors.WithMessage(err, notFoundErr), logData)
+							return &taskError{errors.New(notFoundErr), http.StatusNotFound}
+						}
+						log.ErrorCtx(ctx, errors.WithMessage(err, "failed to update build hierarchy task state"), logData)
+						return &taskError{err, http.StatusInternalServerError}
+					}
+				}
+			}
+			if !hasSearchIndexImportTask {
+				validationErrs = append(validationErrs, errors.New("bad request - missing search index task"))
+			}
+		}
+
+		if !hasImportTasks {
+			validationErrs = append(validationErrs, errors.New("bad request - request body does not contain any import tasks"))
+		}
+
+		if len(validationErrs) > 0 {
+			for _, err := range validationErrs {
+				log.ErrorCtx(ctx, errors.WithMessage(err, "validation error"), logData)
+			}
+			// todo: add all validation errors to the response
```

Comment:
no time like the present :)

Reply:
hah, i agree but as this is just code I've moved, im gonna go with not in this PR :)